Automattic/newspack-plugin
# feat: integrate newspack-image-credits into main plugin and wizard UI

### All Submissions:






### Changes proposed in this Pull Request:



Integrates the separate [Newspack Image Credits plugin](https://github.com/Automattic/newspack-image-credits) into this plugin, which will render the separate plugin obsolete for Newspack sites.

Instead of showing the settings in the Media settings page, shows them in the Newspack Site Design > Settings wizard UI.

Also makes a minor stylistic change to the Image Uploader component to display the selected image at the proper scale instead of as a tiny background thumbnail. cc @thomasguillot to review/test this change.

**Note:** I toyed with the idea of refactoring the options that store these settings into proper theme mods so they can also be updated in the Customizer, like the other site design settings. But I worry that might cause disruption for sites that rely on the separate plugin, which uses options. I'm not sure refactoring is worth it—thoughts?

### How to test the changes in this Pull Request:

1. First, install and activate the [Newspack Image Credits plugin](https://github.com/Automattic/newspack-image-credits) if not already installed. Follow testing instructions in https://github.com/Automattic/newspack-image-credits/pull/9 so that you have some pre-existing images with test credits and settings before you test this PR.
2. Check out this branch, rebuild, and deactivate the Newspack Image Credits plugin.
3. Confirm that the images and settings you tested in step 1 continue to work as they did prior to checking out this branch.
4. Go to **WP admin > Newspack > Site Design > Settings**. Confirm there's a new section here with the Media Credits settings:

<img width="1062" alt="Screen Shot 2021-09-17 at 5 21 45 PM" src="https://user-images.githubusercontent.com/2230142/133863173-e341803a-355f-471a-88af-410a7d1e6b4b.png">

5. Play with the settings here and save. Confirm that your settings save as expected, even after refreshing the page. Also confirm that your saved settings are reflected on the front-end as expected.
6. Reactivate the Newspack Image Credits plugin and confirm that having both plugins active won't cause any trouble in the admin or on the front-end (so we won't have to worry about immediately deactivating this plugin on all sites that might have it).

### Other information:
